### PR TITLE
[Enhancement] Simple compaction scheduler for lake tables

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 128 / 296 (43.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|105|00.00%|[41, 43, 46, 47, 51, 52, 53, 54, 56, 57, 58, 59, 62, 63, 64, 73, 74, 75, 76, 77, 79, 80, 81, 82, 83, 86, 89, 90, 93, 94, 95, 96, 97, 101, 102, 104, 105, 106, 107, 110, 111, 112, 113, 115, 116, 117, 118, 119, 120, 121, 125, 126, 127, 128, 129, 130, 132, 133, 134, 136, 137, 138, 139, 140, 142, 143, 144, 146, 147, 148, 149, 151, 153, 154, 155, 156, 157, 158, 160, 162, 163, 164, 165, 167, 168, 169, 172, 173, 174, 175, 176, 177, 178, 179, 181, 182, 183, 184, 190, 191, 192, 194, 195, 196, 198]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[483, 590, 591, 1053, 1054]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|41|59|69.49%|[45, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 64, 65, 67, 80, 93]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByCount.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByTime.java|19|19|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 128 / 299 (42.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|108|00.00%|[41, 42, 50, 51, 55, 56, 57, 59, 60, 61, 62, 64, 65, 66, 67, 70, 71, 72, 81, 82, 83, 84, 85, 87, 88, 89, 90, 91, 94, 97, 98, 101, 102, 103, 104, 105, 109, 110, 112, 113, 114, 115, 117, 119, 120, 121, 122, 124, 125, 126, 127, 129, 131, 135, 136, 137, 138, 139, 140, 142, 143, 144, 146, 147, 148, 149, 150, 152, 153, 154, 156, 157, 158, 159, 161, 163, 164, 165, 166, 167, 168, 170, 172, 173, 174, 175, 177, 178, 179, 182, 183, 184, 185, 186, 187, 188, 189, 191, 192, 193, 194, 200, 201, 202, 204, 205, 206, 208]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[483, 590, 591, 1053, 1054]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|41|59|69.49%|[45, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 64, 65, 67, 80, 93]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByCount.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByTime.java|19|19|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CompactionManager {
+    private static final Logger LOG = LogManager.getLogger(CompactionManager.class);
+
+    private final Map<PartitionIdentifier, ListNode2<PartitionStatistics>> statisticsMap = new HashMap<>();
+    private final DoublyLinkedList<PartitionStatistics> statisticsList = new DoublyLinkedList<>();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java:21]: Better to use priority queue.

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.lake.compaction;
+
+import com.clearspring.analytics.util.DoublyLinkedList;
+import com.clearspring.analytics.util.ListNode2;
+import com.clearspring.analytics.util.Lists;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java:7]:
```suggestion
import com.google.common.collect.Lists;
```

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CompactionManager {
+    private static final Logger LOG = LogManager.getLogger(CompactionManager.class);
+
+    private final Map<PartitionIdentifier, ListNode2<PartitionStatistics>> statisticsMap = new HashMap<>();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java:20]: Why use `ListNode2<T>`, not directly use `PartitionStatistics`?

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            return;
+        }
+        Database db = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb(partitionIdentifier.getDbId());
+        if (db == null) {
+            compactionManager.removePartition(partitionIdentifier);
+            return;
+        }
+
+        if (!db.tryReadLock(50, TimeUnit.MILLISECONDS)) {
+            LOG.info("Skipped partition compaction due to get database lock timeout");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java:71]: Should recover `doingCompaction` state.

File: fe/fe-core/src/main/java/com/starrocks/lake/ShardDeleter.java [hunk 5]
```diff
@@ -1,8 +1,8 @@
 // This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
 
-
 package com.starrocks.lake;
 
+import com.clearspring.analytics.util.Lists;
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/ShardDeleter.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        CompactionManager compactionManager = GlobalStateMgr.getCurrentState().getCompactionManager();
+        PartitionIdentifier partitionIdentifier = compactionManager.choosePartitionToCompact();
+        if (partitionIdentifier == null) {
+            try {
+                Thread.sleep(IDLE_LOOP_INTERVAL_MS - BUSY_LOOP_INTERVAL_MS);
+            } catch (InterruptedException ignored) {
+            }
+            return;
+        }
+        Database db = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb(partitionIdentifier.getDbId());
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            LOG.debug("Finished loading: {}", listNode.getValue());
+        }
+    }
+
+    public synchronized void handleCompactionFinished(PartitionIdentifier partition, long version) {
+        // FE's follower nodes may have a different timestamp with the leader node.
+        long now = System.currentTimeMillis();
+        ListNode2<PartitionStatistics> listNode = statisticsMap.get(partition);
+        if (listNode == null) {
+            PartitionStatistics statistics = new PartitionStatistics(partition, version, now, version);
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+import java.util.concurrent.TimeUnit;
+
+public class CompactionDispatchDaemon extends MasterDaemon {
+    private static final Logger LOG = LogManager.getLogger(CompactionDispatchDaemon.class);
+    private static final String HOST_NAME = FrontendOptions.getLocalHostAddress();
+    private static final long TXN_TIMEOUT_SECOND = 1800L;
+    private static final long BUSY_LOOP_INTERVAL_MS = 10L;
+    private static final long IDLE_LOOP_INTERVAL_MS = 500L;
+    private static final long MIN_COMPACTION_INTERVAL_ON_SUCCESS = 1000L;
+    private static final long MIN_COMPACTION_INTERVAL_ON_FAILURE = 5000L;
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java (hunk 3 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java (hunk 4 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java:71]: updated

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionManager.java]: The PriorityQueue in JDK does not support changing priority.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 124 / 293 (42.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|109|00.00%|[41, 42, 50, 51, 55, 56, 57, 59, 60, 61, 62, 64, 65, 66, 67, 70, 71, 72, 73, 82, 83, 84, 85, 86, 88, 89, 90, 91, 92, 95, 98, 99, 102, 103, 104, 105, 106, 110, 111, 113, 114, 115, 116, 118, 120, 121, 122, 123, 125, 126, 127, 128, 130, 132, 136, 137, 138, 139, 140, 141, 143, 144, 145, 147, 148, 149, 150, 151, 153, 154, 155, 157, 158, 159, 160, 162, 164, 165, 166, 167, 168, 169, 171, 173, 174, 175, 176, 178, 179, 180, 183, 184, 185, 186, 187, 188, 189, 190, 192, 193, 194, 195, 201, 202, 203, 205, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[483, 590, 591, 1053, 1054]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|37|52|71.15%|[40, 46, 47, 48, 49, 50, 52, 53, 54, 55, 57, 58, 60, 73, 86]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByCount.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionPickerByTime.java|19|19|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 124 / 293 (42.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|109|00.00%|[41, 42, 50, 51, 55, 56, 57, 59, 60, 61, 62, 64, 65, 66, 67, 70, 71, 72, 73, 82, 83, 84, 85, 86, 88, 89, 90, 91, 92, 95, 98, 99, 102, 103, 104, 105, 106, 110, 111, 113, 114, 115, 116, 118, 120, 121, 122, 123, 125, 126, 127, 128, 130, 132, 136, 137, 138, 139, 140, 141, 143, 144, 145, 147, 148, 149, 150, 151, 153, 154, 155, 157, 158, 159, 160, 162, 164, 165, 166, 167, 168, 169, 171, 173, 174, 175, 176, 178, 179, 180, 183, 184, 185, 186, 187, 188, 189, 190, 192, 193, 194, 195, 201, 202, 203, 205, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[483, 590, 591, 1053, 1054]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|37|52|71.15%|[40, 46, 47, 48, 49, 50, 52, 53, 54, 55, 57, 58, 60, 73, 86]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByTime.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByCount.java|14|14|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+        TransactionState.LoadJobSourceType loadJobSourceType = TransactionState.LoadJobSourceType.LAKE_COMPACTION;
+        TransactionState.TxnSourceType txnSourceType = TransactionState.TxnSourceType.FE;
+        TransactionState.TxnCoordinator coordinator = new TransactionState.TxnCoordinator(txnSourceType, HOST_NAME);
+
+        long txnId;
+        try {
+            txnId = GlobalStateMgr.getCurrentGlobalTransactionMgr().beginTransaction(db.getId(),
+                    Lists.newArrayList(table.getId()), label, coordinator,
+                    loadJobSourceType, TXN_TIMEOUT_SECOND);
+        } catch (BeginTransactionException | AnalysisException | LabelAlreadyUsedException | DuplicatedRequestException e) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionDispatchDaemon.java:113]: Should recover `doingCompaction` state.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 124 / 294 (42.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|110|00.00%|[41, 42, 50, 51, 55, 56, 57, 59, 60, 61, 62, 64, 65, 66, 67, 70, 71, 72, 73, 82, 83, 84, 85, 86, 88, 89, 90, 91, 92, 95, 98, 99, 102, 103, 104, 105, 106, 110, 111, 113, 114, 115, 116, 117, 119, 121, 122, 123, 124, 126, 127, 128, 129, 131, 133, 137, 138, 139, 140, 141, 142, 144, 145, 146, 148, 149, 150, 151, 152, 154, 155, 156, 158, 159, 160, 161, 163, 165, 166, 167, 168, 169, 170, 172, 174, 175, 176, 177, 179, 180, 181, 184, 185, 186, 187, 188, 189, 190, 191, 193, 194, 195, 196, 202, 203, 204, 206, 207, 208, 210]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[483, 590, 591, 1053, 1054]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|37|52|71.15%|[40, 46, 47, 48, 49, 50, 52, 53, 54, 55, 57, 58, 60, 73, 86]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByTime.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByCount.java|14|14|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 124 / 293 (42.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|109|00.00%|[41, 42, 50, 51, 55, 56, 57, 59, 60, 61, 62, 64, 65, 66, 67, 70, 71, 72, 73, 82, 83, 84, 85, 86, 88, 89, 90, 91, 92, 95, 98, 99, 102, 103, 104, 105, 106, 110, 111, 113, 114, 115, 116, 118, 120, 121, 122, 123, 125, 126, 127, 128, 130, 132, 136, 137, 138, 139, 140, 141, 143, 144, 145, 147, 148, 149, 150, 151, 153, 154, 155, 157, 158, 159, 160, 162, 164, 165, 166, 167, 168, 169, 171, 173, 174, 175, 176, 178, 179, 180, 183, 184, 185, 186, 187, 188, 189, 190, 192, 193, 194, 195, 201, 202, 203, 205, 206, 207, 209]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|7|00.00%|[30, 35, 38, 39, 40, 41, 43]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|3|14|21.43%|[39, 43, 47, 57, 58, 63, 67, 69, 70, 71, 72]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|7|28.57%|[482, 600, 601, 1060, 1061]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionIdentifier.java|6|16|37.50%|[21, 25, 29, 34, 35, 37, 38, 40, 41, 51]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|37|52|71.15%|[40, 46, 47, 48, 49, 50, 52, 53, 54, 55, 57, 58, 60, 73, 86]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|16|22|72.73%|[60, 86, 87, 89, 90, 91]|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|3|4|75.00%|[106]|
|:large_blue_circle:|com/starrocks/lake/compaction/PartitionStatistics.java|21|26|80.77%|[49, 50, 57, 58, 74]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByTime.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionPickerByCount.java|14|14|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**